torch/ao/quantization/fx/fuse.py

```diff
@@ -75,10 +78,10 @@ def get_root_node(node_pattern):
                 node_subpattern = None
             if maybe_last_node is node:
                 assert obj is not None
-                # TODO: currently we hard code the root node, which only works for
-                # a sequence of ops and assume the root node is the last node,
-                # we want to make this more general to support more complex patterns
-                root_node = get_root_node(matched_node_pattern)  # type: ignore[index]
+                root_node_getter = fusion_pattern_to_root_node_getter.get(pattern, default_root_node_getter)
+                root_node = root_node_getter(matched_node_pattern)  # type: ignore[index]
```

Comment:
should this have a check that root_node exists?

Reply:
we are inside the branch where we have a match I think, so we don't need to check if it exists, but we can have a validation that the root_node has the same type that is specified in the pattern, e.g. torch.nn.Conv2d